src/azure-cli/azure/cli/command_modules/cosmosdb/commands.py

```diff
@@ -165,7 +165,7 @@ def load_command_table(self, _):
         g.command('list-connection-strings', 'list_connection_strings', table_transformer=list_connection_strings_output, deprecate_info=g.deprecate(redirect='cosmosdb keys list --type connection-strings', hide=True))
         g.custom_command('regenerate-key', 'cli_cosmosdb_regenerate_key', deprecate_info=g.deprecate(redirect='cosmosdb keys regenerate', hide=True))
         g.command('check-name-exists', 'check_name_exists')
-        g.command('delete', 'begin_delete', confirmation=True)
+        g.command('delete', 'begin_delete', confirmation=True, supports_no_wait=True)
```

Comment:
```suggestion
        g.custom_command('delete', 'cli_cosmosdb_delete', confirmation=True, supports_no_wait=True)
```
To support `--no-wait`, you need to add the no wait logic in your custom func (let say in `cli_cosmosdb_delete`), call `client.begin_delete` in `sdk_no_wait` func.

Take this as an example: https://github.com/Azure/azure-cli/blob/168206d11be6b14e692e3b733ad54a261e5832d6/src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py#L2326-L2338

Reply:
I was searching for the begin delete custom function but couldn't find it. I was aware things should be wrapped but after I couldn't find it I figured begin delete must be some generic function. I'll give it another look

Reply:
@evelyn-ys I pushed an update. Please be patient with me. I'm very new to this whole repository and probably doing things horribly wrong. Thanks for your assistance!

Reply:
`begin_delete` should be the interface defined in `azure-mgmt-cosmosdb` SDK which Azure CLI cosmosdb module depends on.

If you are new to CLI development, better to read the doc first https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md